release.md

```diff
@@ -0,0 +1,63 @@
+# Release Processes
+
+As per the [Versioning and release process of the libs artifacts](https://github.com/falcosecurity/libs/blob/master/proposals/20210524-versioning-and-release-of-the-libs-artifacts.md), this repository includes different groups of artifacts that needs to be versioned and released. Moreover, those artifacts follows different versioning schemas (see the [Driver SemVer](https://github.com/falcosecurity/libs/blob/master/proposals/20210818-driver-semver.md) and [Versioning schema](https://github.com/falcosecurity/libs/blob/master/proposals/20220203-versioning-schema-amendment.md)) proposals).
+
+We have two separate release processes which occur **independently** of each other: 
+- The **drivers release**
+- The **libs release** (ie. _libsinp_ and _libscap_)
+
+The drivers release process is mainly automated and happens outside this repository. The actual building and distribution system is implemented in our [test-infra](https://github.com/falcosecurity/test-infra) by the [Driverkit Build Grid](https://github.com/falcosecurity/test-infra/tree/master/driverkit), and drivers are published to https://download.falco.org/?prefix=driver/. The [drivers versioning](#Drivers-versioning) process happens in this repo. 
+
+>_Note that not all versioned releases will be built and distributed. Drivers distribution is indeed implemented and maintained only to satisfy Falco's needs._
+
+The libs release process is currently under development and limited to the versioning process only. The [libs versioning](#Libs-versioning) process happens in this repo.
+
+
+## Release procedure
+
+Regardless if it is a driver or a libs release when initiating a new release, we do the following process:
+
+1. We decide together (usually in the #falco channel in [slack](https://kubernetes.slack.com/messages/falco)) if the source code is a good shape and if it's the case to be released
+2. We double-check if the versioning rules have been respected (see sections below), then we pick the next version number to tag (i.e., a _git tag_)
+3. A person with repository rights creates a [new release using the GitHub UI](https://github.com/falcosecurity/libs/releases/new) (a git tag will be automatically created)
+
+> _At the time of writing, no other steps are needed since the whole release process is still in development. This document will be updated once the definitive process is fully implemented._
+
+
+## Drivers versioning
+
+The *driver version number* represents the build version of kernel-space drivers (i.e., the kernel module, the eBPF probe, and possibly any other kernel-space drivers).
+
+**Requirements**
+
+- The version MUST be a [SemVer 2.0](https://semver.org/spec/v2.0.0.html) string.
+
+- Since our driver APIs are assumed to be stable, the major version number MUST be equal to or greater than `1`.
+
+- The version string MUST be suffixed with `+driver` to distinguish it from libs version numbers.
```

Comment:
Since both drivers releases and libs releases will be tagged in the same repository, our plan is to use the `+driver` suffix to distinguish between the two sets of git tags. The `+` sign is intentional since it denotes the beginning of [build metadata] (which does not change the version precedence, as per semver specs).

In the end, we expect to have something like:

```shell
$ git tag
1.0.0+driver
0.55.0
1.1.0+driver
0.56.0
0.56.1
...and so on
```

PS I don't like the `+driver` suffix too much, but I can't find anything better. Any suggestion?

Reply:
@leogr how about using a prefix instead? Something like `driver-x.y.z` for the driver and `libs-x.y.z` for the userspace components.
That's the the approach I've used in the past, in cases like this, where a single git repo would essentially host functionally related components which would however follow separate versioning and release processes.
We might also leverage `git describe`'s `--match` option:
```
$ git tag driver-0.0.1
$ git describe --tags --match 'driver-*' | cut -c 8-
0.0.1
```
Notice how the above will however NOT provide an exact tag (as I for one would expect) when later commits have changed something NOT in `driver/`, but with some tweaking we might get the same result (if needed, I'm not quite sure of all the possible use cases). 
I assume moving the driver to its own, separate repo is off the table, right?

Reply:
Hey @iurly 

Your observation is correct. I also used that approach in the past, and we are using it in our [charts](https://github.com/falcosecurity/charts). However, there are reasons why we decided to do it differently here.

I will try to explain it here for the records :smile_cat: 

Driverkit, test-infra, and other tools in our CI assume that the driver's version is precisely equal to a _[git ref](https://git-scm.com/book/en/v2/Git-Internals-Git-References)_ in this repository (so a git commit, a git branch, or a git tag are valid driver versions). Note that this approach is also helpful during development time: one could use driverkit to build a driver from a branch, for example. We still want to keep it. Moreover, changing that assumption would require considerable effort.

Now we are adding SemVer to the equation. Unfortunately, `driver-x.y.z` is not a compliant SemVer string. In general, we can't use prefixes for this reason. 

Considering the two requirements, I suggested working around the issue by using SemVer build metadata. Also, we use the `+driver` suffix only for the driver versions because this repository is named _libs_, so it would be redundant to use a suffix for the libs versions too.

I hope my explanation was adequate.

PS
Transfering the `driver/` in its repo is currently an option. We already have an accepted proposal that mandates us to keep drivers here. Maybe in the future!